core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -512,6 +512,44 @@ class RequestChannel(val queueSize: Int,
 
 }
 
+class RequestBreakdownMetrics(val name: String) extends KafkaMetricsGroup {
+  val HistogramMetricType = "histogram"
+  val DurationDataType = "duration"
+  val tags = Map("request" -> name)
+  // This is to make writing MBean query easier
+  val tagsForHistograms = tags ++ Map("metricType" -> HistogramMetricType)
+  val tagsForDurationHistograms = tagsForHistograms ++ Map("metricType" -> HistogramMetricType, "dataType" -> DurationDataType)
+}
+
+object LiCombinedControlRequestBreakdownMetrics {
+  import RequestMetrics.{LocalTimeMs, RequestQueueTimeMs}
+
+  private val metrics = new RequestBreakdownMetrics(ApiKeys.LI_COMBINED_CONTROL.name)
+  private val breakdownRequestQueueTime = mutable.Map[ApiKeys, Histogram]()
+  private val breakdownLocalTime = mutable.Map[ApiKeys, Histogram]()
+
+  // Initialize request that we want breakdown time metrics
+  Seq(
+    ApiKeys.LEADER_AND_ISR,
+    ApiKeys.UPDATE_METADATA,
+    ApiKeys.STOP_REPLICA
+  ).foreach{apiKey =>
+    val decomposedQueueTimeName = s"decomposed-${apiKey.name}-${RequestQueueTimeMs}"
```

Comment:
Maybe call this the DelayedTime instead of QueueTime, since for the other requests, their QueueTime is only the time they spent in the Request Queue.
But here the DelayedTime = QueueTime + waiting time for other parts of the LiComibinedControl to finish.

Reply:
Fixed